### PR TITLE
[LA][AOSS-1426] Consume data in the format returned by agent-client-a…

### PR DESCRIPTION
**Pull request by bravesirrobin81**
Doesn't yet use new URLs

### CODE REVIEW

File: app/uk/gov/hmrc/agentclientauthorisation/model/ClientRequestStatusViewData.scala [hunk 1]
```diff
@@ -20,7 +20,18 @@ import org.joda.time.DateTime
 import play.api.libs.json.Json
 
 case class ClientRequestStatusViewData(requests: List[ClientRequestStatus])
-case class ClientRequestStatus(clientFullName:String, requestDate:DateTime, status: AuthorisationStatus = Pending)
+case class RequestStatusChange(status: AuthorisationStatus, changeTime: DateTime)
+case class ClientRequestStatus(clientFullName:String, requestDate:DateTime, status: List[RequestStatusChange]) {
+  def currentStatus: AuthorisationStatus = {
+    status match {
+      case head :: rest => head.status
```
Comment (szmg) [inline app/uk/gov/hmrc/agentclientauthorisation/model/ClientRequestStatusViewData.scala]: That's the other way around -- new status change events are pushed to the array of events, thus the most recent is at the end.

File: test/uk/gov/hmrc/agentclientauthorisation/acceptance/stubs/AgentClientAuthorisationStub.scala [hunk 2]
```diff
@@ -94,7 +94,10 @@ object AgentClientAuthorisationStub {
       |  "clientFullName": "${request.clientFullName}",
       |  "requestDate": ${request.requestDate.getMillis},
       |  "regime": "${request.regime}",
-      |  "status": "${request.status}"
+      |  "status": [{
+      |    "status": "${request.status}",
+      |    "changeTime": ${request.requestDate.getMillis}
```
Comment (szmg) [inline test/uk/gov/hmrc/agentclientauthorisation/acceptance/stubs/AgentClientAuthorisationStub.scala]: This is called `time` in the backend, as well as in the message. Do you think it should be changed there?

Comment (bravesirrobin81) [inline test/uk/gov/hmrc/agentclientauthorisation/acceptance/stubs/AgentClientAuthorisationStub.scala]: Better to change it here, I think.

File: app/uk/gov/hmrc/agentclientauthorisation/model/ClientRequestStatusViewData.scala (hunk 1 above)
Comment (bravesirrobin81) [inline app/uk/gov/hmrc/agentclientauthorisation/model/ClientRequestStatusViewData.scala]: Ah, I'd had the impression from @howzat that the ordering was most recent first. I'll change this.

File: app/uk/gov/hmrc/agentclientauthorisation/model/ClientRequestStatusViewData.scala [hunk 3]
```diff
@@ -20,7 +20,18 @@ import org.joda.time.DateTime
 import play.api.libs.json.Json
 
 case class ClientRequestStatusViewData(requests: List[ClientRequestStatus])
-case class ClientRequestStatus(clientFullName:String, requestDate:DateTime, status: AuthorisationStatus = Pending)
+case class RequestStatusChange(status: AuthorisationStatus, changeTime: DateTime)
+case class ClientRequestStatus(clientFullName:String, requestDate:DateTime, status: List[RequestStatusChange]) {
```
Comment (szmg) [inline app/uk/gov/hmrc/agentclientauthorisation/model/ClientRequestStatusViewData.scala]: `requestDate` is not directly part of the request, but its first status change event. (Thus it's not in the response like this, I think the connector needs to be changed.)

Comment (bravesirrobin81) [inline app/uk/gov/hmrc/agentclientauthorisation/model/ClientRequestStatusViewData.scala]: Fixed.

**Pull request merged**